standard/links_stream.lua

```diff
@@ -83,45 +80,133 @@ function StreamLinks.processStreams(forwardedInputArgs)
-			if platformName ~= 'stream' then
-				local lookUpPlatform = StreamLinks.streamPlatformLookupNames[platformName] or platformName
+	local processedStreams = {}
+	Array.forEach(StreamLinks.countdownPlatformNames, function(platformName)
+		Table.mergeInto(processedStreams, StreamLinks._processStreamsOfPlatform(streams, platformName))
+	end)
+
+	return processedStreams
+end
 
-				streamValue = StreamLinks.resolve(lookUpPlatform, streamValue)
+---@param streamValues {[string]: string}
+---@param platformName string
+---@return {[string]: string}
+function StreamLinks._processStreamsOfPlatform(streamValues, platformName)
+	local platformStreams = {}
+	local legacyStreams = {}
+	local enCounter = 0
+
+	for key, streamValue in Table.iter.spairs(streamValues) do
+		if platformName ~= TLNET_STREAM then
+			streamValue = StreamLinks.resolve(platformName, streamValue)
+		end
+
+		-- legacy key
+		if key:match('^' .. platformName .. '%d*$') then
+			table.insert(legacyStreams, streamValue)
+		elseif key:match('^' .. platformName .. '_%a+_%d+') then
+			local streamKey = StreamLinks.StreamKey(key)
+			if streamKey.languageCode == 'en' then
+				enCounter = enCounter + 1
 			end
+			platformStreams[streamKey:toString()] = streamValue
+		end
+	end
 
-			local key = StreamLinks.StreamKey(platformName):toString()
-			streams[key] = streamValue
-			streams[platformName] = streamValue -- Legacy
+	for _, streamValue in ipairs(legacyStreams) do
+		if not Table.includes(platformStreams, streamValue) then
+			enCounter = enCounter + 1
+			local streamKey = StreamLinks.StreamKey(platformName, 'en', enCounter):toString()
+			platformStreams[streamKey] = streamValue
+			platformStreams[platformName] = streamValue -- Legacy
 		end
 	end
 
+	if Logic.isEmpty(platformStreams) then
+		platformStreams = {[platformName] = Variables.varDefault(platformName)}
+	end
+
+	return platformStreams
+end
+
+---@param platform string
+---@param streamValue string
+---@return string?
+function StreamLinks.displaySingle(platform, streamValue)
+	local icon = '<i class="lp-icon lp-icon-21 lp-' .. platform .. '"></i>'
+	if platform == TLNET_STREAM then
+		return Page.makeExternalLink(icon, 'https://tl.net/video/streams/' .. streamValue)
+	end
+
+	local streamLink = StreamLinks.resolve(platform, streamValue)
+	if not streamLink then return nil end
+
+	return Page.makeInternalLink({}, icon, 'Special:Stream/' .. platform .. '/' .. streamValue)
+end
+
+---@param streams {string: string[]}
+---@param options {addSpace: boolean?}?
+---@return string?
+function StreamLinks.display(streams, options)
+	options = options or {}
+	local displays = {}
+	Array.forEach(StreamLinks.countdownPlatformNames, function(platform)
+		Array.forEach(streams[platform] or {}, function(streamValue)
+			table.insert(displays, StreamLinks.displaySingle(platform, streamValue))
+		end)
+	end)
+	if Table.isEmpty(displays) then return nil end
+	return table.concat(displays, options.addSpace and ' ' or nil)
+end
```

Comment:
Actually, would it not be better to return an array, and the caller can make the option if adding space, or some adding padding (new line, icon, bullet)?